src/NovoNordisk.CriticalPath/Graph.cs

```diff
@@ -0,0 +1,119 @@
+using NovoNordisk.CriticalPath.Exceptions;
+
+namespace NovoNordisk.CriticalPath;
+
+internal class Graph
+{
+    private readonly ISet<Activity> _nodes;
+    private readonly Dictionary<Activity, List<Activity>> _edgesFrom;
+    private readonly Dictionary<Activity, List<Activity>> _edgesTo;
+    
+    private Graph(ISet<Activity> nodes, Dictionary<Activity, List<Activity>> edgesFrom, Dictionary<Activity, List<Activity>> edgesTo)
```

Comment:
Have you considered to spliut the unit tests now the Critical Path method has been refactored and a new class created?

Reply:
I have decided not to do so. The public API is still the critical path method - the Graph class is just a supporting class, not to be used by itself. The class is also internal, so I cannot access it from the test project (unless using InternalsVisibleTo).